packages/system/src/utils/store.ts

```diff
@@ -145,64 +157,72 @@ function calculateXYZPosition<NodeType extends NodeBase>(
   );
 }
 
-export function handleParentExpand(
-  nodes: InternalNodeBase[],
-  nodeLookup: NodeLookup
+type ExpandParentNode<NodeType> = NodeType & { parentId: string; expandParent: true };
+
+export function handleExpandParent(
+  nodes: ExpandParentNode<InternalNodeBase>[],
```

Comment:
Let's change the API here and use something like `[{ parentId, rect }]` instead of nodes